### PR TITLE
chore: update appveyor image

### PR DESCRIPTION
**Pull request by mndeveci**
Updating Appveyor Linux image to `Ubuntu2204`
- Install `jdk21` as it is not available in Appveyor images yet
- Update `maven` and `gradle` installations to comply with jdk21 update abpove
- Adding `python3.12` runtime information to `PATH` environment variable so that we can build lambda functions with that runtime.
- Limit test cases when running ARM64 and building inside container for java runtimes. For gradlew and gradle-kotlin test cases, we only run them if java runtime is latest (`java21` atm).
- Remove `provided` (al1) test cases for Go and Rust since these runtimes are expected to run on `provided.al2` and `provided.al2023` runtimes.
- Revert `Makefile` and bring back `Makefile-container` which uses available python executables for each environment.

Update Appveyor Windows image:
- install Corretto JDK 21 and set corresponding
- Upgrade gradle
- set path for Python3.12
- skip all build command tests for the new runtimes (`provided.al2023`, `python3.12`, `nodejs20.x` and `java21`) in AppVeyor Windows that require docker. This is due to the docker version in AppVeyor Windows are not new enough (`>=20.10.10`) to support AL2023-based containers. See: https://docs.docker.com/engine/release-notes/20.10/#201010

By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0).

### CODE REVIEW

File: appveyor-linux-binary.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 version: 1.0.{build}
 image:
-  - Ubuntu2004
+  - Ubuntu2204
```
Comment (hawflau) [inline appveyor-linux-binary.yml:3]:
```suggestion
  - Ubuntu2004
```

File: appveyor-ubuntu.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 version: 1.0.{build}
 image: 
-  - Ubuntu2004
+  - Ubuntu2204
```
Comment (hawflau) [inline appveyor-ubuntu.yml:3]:
```suggestion
  - Ubuntu2004
```

File: appveyor-ubuntu.yml [hunk 3]
```diff
@@ -33,18 +33,29 @@ install:
...
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12
+  - sh: python3.12 -V
+  - sh: python3.12 -m venv $HOME/venv3.12
```
Comment (hawflau) [inline appveyor-ubuntu.yml:49]:
```suggestion
  - sh: which python3.12
```

File: appveyor-ubuntu.yml [hunk 4]
```diff
@@ -33,18 +33,29 @@ install:
...
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12
+  - sh: python3.12 -V
+  - sh: which python3.12
```
Comment (hawflau) [inline appveyor-ubuntu.yml:49]:
```suggestion
  - sh: which python3.12
  - sh: $HOME/venv3.12/bin/python -V
```

File: appveyor-ubuntu.yml [hunk 5]
```diff
@@ -33,18 +33,30 @@ install:
...
+  - sh: echo "deb [signed-by=/usr/share/keyrings/corretto-keyring.gpg] https://apt.corretto.aws stable main" | sudo tee /etc/apt/sources.list.d/corretto.list
+  - sh: sudo apt-get update; sudo apt-get install -y java-21-amazon-corretto-jdk
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12
```
Comment (hawflau) [inline appveyor-ubuntu.yml:47]:
```suggestion
  - sh: sudo apt update; sudo apt install -y python3.12 python3.12-venv
```

File: appveyor-ubuntu.yml [hunk 6]
```diff
@@ -33,18 +33,30 @@ install:
...
+  - sh: python3.12 -V
+  - sh: which python3.12
```
Comment (hawflau) [inline appveyor-ubuntu.yml:49]:
```suggestion
  - sh: python3.12 -m venv $HOME/venv3.12
```

File: appveyor-ubuntu.yml [hunk 7]
```diff
@@ -33,18 +33,30 @@ install:
...
+  - sh: sudo apt-get update; sudo apt-get install -y java-21-amazon-corretto-jdk
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12 python3.12-venv
+  - sh: python3.12 -V
```
Comment (hawflau) [inline appveyor-ubuntu.yml:48]:
```suggestion
  - sh: python3.12 -V
  - sh: python3.12 -m pip install --upgrade pip
```

File: appveyor-ubuntu.yml [hunk 8]
```diff
@@ -33,18 +33,31 @@ install:
...
+  - sh: echo "deb [signed-by=/usr/share/keyrings/corretto-keyring.gpg] https://apt.corretto.aws stable main" | sudo tee /etc/apt/sources.list.d/corretto.list
+  - sh: sudo apt-get update; sudo apt-get install -y java-21-amazon-corretto-jdk
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12 python3.12-venv
```
Comment (hawflau) [inline appveyor-ubuntu.yml:47]:
```suggestion
  - sh: sudo apt update; sudo apt install -y python3.12 python3.12-venv python3.12-distutils
```

File: appveyor-windows.yml [hunk 9]
```diff
@@ -53,23 +53,27 @@ cache:
...
   - "mvn --version"
 
   # Make sure the temp directory exists for Python to use.
   - ps: "mkdir -Force C:\\tmp"
-  - 'set PATH=%PYTHON_HOME%;C:\Ruby32-x64\bin;%PATH%;C:\Python37-x64;C:\Python39-x64;C:\Python310-x64;C:\Python311-x64'
+  - 'set PATH=%PYTHON_HOME%;C:\Ruby32-x64\bin;%PATH%;C:\Python37-x64;C:\Python39-x64;C:\Python310-x64;C:\Python311-x64;C:\Python312-x64'
   - "echo %PYTHON_HOME%"
   - "echo %PATH%"
   - "python --version"
+  - "choco upgrade docker-engine"
```
Comment (hawflau) [inline appveyor-windows.yml:76]:
```suggestion
  - "choco upgrade docker-engine"
  - ps: "Restart-Service docker"
```

File: appveyor-ubuntu.yml [hunk 10]
```diff
@@ -33,18 +33,31 @@ install:
...
+  - sh: JAVA_HOME=/usr/lib/jvm/java-21-amazon-corretto
+  - sh: PATH=$JAVA_HOME/bin:$PATH
+  - sh: java --version
+  - sh: javac --version
+
+  # install Python3.12
+  - sh: sudo add-apt-repository ppa:deadsnakes/ppa
+  - sh: sudo apt update; sudo apt install -y python3.12 python3.12-venv python3.12-distutils
+  - sh: python3.12 -V
+  - sh: python3.12 -m pip install --upgrade pip
```
Comment (hawflau) [inline appveyor-ubuntu.yml:49]:
```suggestion
```

File: appveyor-ubuntu.yml [hunk 11]
```diff
@@ -33,18 +33,31 @@ install:
...
+  - sh: python3.12 -m venv $HOME/venv3.12
+  - sh: $HOME/venv3.12/bin/python -V
```
Comment (hawflau) [inline appveyor-ubuntu.yml:51]:
```suggestion
  - sh: $HOME/venv3.12/bin/python -V
  - sh: $HOME/venv3.12/bin/python -m pip install --upgrade pip
```

File: appveyor-windows.yml [hunk 12]
```diff
@@ -53,23 +53,28 @@ cache:
...
+  - ps: "Restart-Service docker"
```
Comment (hawflau) [inline appveyor-windows.yml:77]:
```suggestion
  - sc config docker binpath= "\"C:\Program Files\docker\dockerd.exe\" --run-service -H tcp://0.0.0.0:2375"
  - ps: "Restart-Service docker"
```

File: appveyor-windows.yml [hunk 13]
```diff
@@ -53,23 +53,28 @@ cache:
...
+  - choco upgrade chocolatey
   # setup make
   - "choco install make"
   # setup Java, Maven and Gradle
-  - "choco install gradle -y --force"
-  - 'set JAVA_HOME=C:\Program Files\Java\jdk17'
-  - 'set PATH=%JAVA_HOME%\bin;%PATH%'
-  - "javac -version"
-  - "java -version"
+  - choco install correttojdk --version=21.0.0
```
Comment (hawflau) [inline appveyor-windows.yml:61]:
```suggestion
  - choco install correttojdk --version=21.0.0
  - refreshenv
```

File: appveyor-windows-binary.yml [hunk 14]
```diff
@@ -54,20 +54,22 @@ cache:
...
+  - choco upgrade chocolatey
   # setup make
   - "choco install make"
   # setup Java, Maven and Gradle
-  - "choco install gradle -y --force"
-  - 'set JAVA_HOME=C:\Program Files\Java\jdk17'
-  - 'set PATH=%JAVA_HOME%\bin;%PATH%'
-  - "javac -version"
-  - "java -version"
+  - choco install correttojdk --version=21.0.0
```
Comment (hawflau) [inline appveyor-windows-binary.yml:62]:
```suggestion
  - choco install correttojdk --version=21.0.0
  - refreshenv
```

File: appveyor-windows.yml [hunk 15]
```diff
@@ -53,23 +53,29 @@ cache:
...
+  - refreshenv
```
Comment (hawflau) [inline appveyor-windows.yml:62]:
```suggestion
  - 'set JAVA_HOME=C:\Program Files\Amazon Corretto\jdk21.0.0_35'
  - 'set PATH=%JAVA_HOME%\bin;%PATH%'
```

File: appveyor-windows-binary.yml [hunk 16]
```diff
@@ -54,20 +54,23 @@ cache:
...
+  - refreshenv
```
Comment (hawflau) [inline appveyor-windows-binary.yml:63]:
```suggestion
  - 'set JAVA_HOME=C:\Program Files\Amazon Corretto\jdk21.0.0_35'
  - 'set PATH=%JAVA_HOME%\bin;%PATH%'
```

**Review by hnnasit — APPROVED** [review]

**Review by mildaniel — APPROVED** [review]

File: appveyor-ubuntu.yml [hunk 17]
```diff
@@ -33,21 +33,27 @@ install:
...
-  - sh: "PATH=/opt/mvn/apache-maven-3.8.8/bin:$PATH"
-  - sh: "JAVA_HOME=/usr/lib/jvm/java-17-openjdk-amd64"
-  - sh: "PATH=$JAVA_HOME/bin:$PATH"
-  - sh: "javac -version"
-  - sh: "java -version"
+  - sh: "PATH=/opt/mvn/apache-maven-3.9.5/bin:$PATH"
   - sh: "mvn --version"
 
   - sh: "source ${HOME}/venv${PYTHON_VERSION}/bin/activate"
+  - sh: "rvm reinstall ruby-3.2.2 --with-openssl-dir=/usr/lib/x86_64-linux-gnu" # reinstall ruby3.2 to fix OpenSSL issue
```
Comment (hnnasit) [inline appveyor-ubuntu.yml:56]: Do we need to add this command in `appveyor-linux-binary.yml` as well?

Comment (mndeveci) [reply appveyor-ubuntu.yml:56]: Good point, I think we need to add it there. Let me compare both again and add any missing lines

Comment (hnnasit) [reply appveyor-ubuntu.yml:56]: Thanks Mehmet, new changes LGTM. No other concerns from me.

**Pull request merged**